lib/init.js

```diff
@@ -0,0 +1,141 @@
+'use babel';
+
+// eslint-disable-next-line import/no-extraneous-dependencies, import/extensions
+import { CompositeDisposable } from 'atom';
+
+let path;
+let helpers;
+
+const pattern = '.+:(?<line>\\d+):(?<col>\\d+)-(?<colEnd>\\d+):' +
+  ' \\((?<type>[EW])\\d+\\) (?<message>.*)';
+
+function loadDeps() {
+  if (!helpers) {
+    helpers = require('atom-linter');
+  }
+  if (!path) {
+    path = require('path');
+  }
+}
+
+function checkedOption(params, opt, args) {
+  if (args.length > 0) {
+    params.push(opt);
+    return params.concat(args);
+  }
+
+  return params;
+}
+
+function reportToMessage(report, file) {
+  const end = report.range[1];
+  end[1] += 1;
+
+  return {
+    location: {
+      file,
+      position: [report.range[0], end]
+    },
+    severity: (report.type === 'E') ? 'error' : 'warning',
+    excerpt: report.text
+  };
+}
+
+export default {
+  activate() {
+    this.idleCallbacks = new Set();
+
+    let depCallbackID;
+    const installLinterDeps = () => {
+      this.idleCallbacks.delete(depCallbackID);
+
+      require('atom-package-deps').install('linter-luacheck');
+      loadDeps();
```

Comment:
It will _eventually_, the idea of putting it in here is that Atom will load the dependencies when it is idle (after it's done loading all the other packages).

See comment below for where you need an additional call to `loadDeps`.